src/components/TagWrapper/PTagWrapper.vue

```diff
@@ -161,17 +188,31 @@
   }
 
   function setInlineContainerWidth(tags: HTMLElement[]): void {
-    const totalTagsWidth = tags.reduce((acc, tag) => {
-      const boundingBox = tag.getBoundingClientRect()
+    if (!container.value) {
+      return
+    }
+
+    const totalTagsWidth = getTotalTagsWidth(tags)
+    const computedContainerStyle = getComputedStyle(container.value)
+    const computedContainerGap = computedContainerStyle.getPropertyValue('column-gap')
+    const containerGap = parseFloat(computedContainerGap)
+    const padding = isNaN(containerGap) ? 0 : (tags.length - 1) * containerGap
+
+    const tagsWidthWithPadding = totalTagsWidth + padding
+    container.value.style.width = `${tagsWidthWithPadding}px`
+  }
 
-      return acc + Math.ceil(boundingBox.width)
-    }, 0)
+  function getTotalTagsWidth(tags: HTMLElement[]): number {
+    return Math.ceil(tags.reduce((acc, tag) => {
+      const boundingBox = tag.getBoundingClientRect()
 
-    container.value!.style.width = `${totalTagsWidth}px`
+      return acc + boundingBox.width
+    }, 0))
   }
 
   function createObserver(): void {
-    resizeObserver = new ResizeObserver(calculateOverflow)
+    const throttledCalculateOverflow = throttle(calculateOverflow, 100, { trailing: true })
```

Comment:
I'm surprised we need `throttle`. A resize observer should be tuned to the animation frame. Was that just to fast?

Reply:
Yeah, justttttt a hair too fast; logging the container width was the biggest giveaway - it was always 0 the first time this ran without a throttle